DataFormats/PatCandidates/interface/IsolatedTrack.h

```diff
@@ -22,39 +28,91 @@ namespace pat {
+        explicit IsolatedTrack(const PFIsolation &iso, const PFIsolation &miniiso, float caloJetEm, float caloJetHad,
                                const LorentzVector &p4, int charge, int id,
-                               float dz, float dxy, const PackedCandidateRef &ref) :
+                               float dz, float dxy, float dzError, float dxyError,
+                               const reco::HitPattern &hp, float dEdxS, float dEdxP, int fromPV, int tkQual,
+                               const std::vector<EcalChannelStatusCode> &ecalst,
+                               const std::vector<HcalChannelStatus> & hcalst, int dEta, int dPhi,
+                               const PackedCandidateRef &pcref) :
           LeafCandidate(charge, p4, Point(0.,0.,0.), id),
-          pfIsolationDR03_(iso),
-          miniIsolation_(miniiso),
-          dz_(dz), dxy_(dxy),
-          packedCandRef_(ref) {}
+          pfIsolationDR03_(iso), miniIsolation_(miniiso), 
+          matchedCaloJetEmEnergy_(caloJetEm), matchedCaloJetHadEnergy_(caloJetHad),
+          dz_(dz), dxy_(dxy), dzError_(dzError), dxyError_(dxyError),
+          fromPV_(fromPV), trackQuality_(tkQual), dEdxStrip_(dEdxS), dEdxPixel_(dEdxP), 
+          hitPattern_(hp), 
+          crossedEcalStatus_(ecalst), crossedHcalStatus_(hcalst),
+          deltaEta_(dEta), deltaPhi_(dPhi),
+          packedCandRef_(pcref) {}
 
         ~IsolatedTrack() {}
 
         const PFIsolation& pfIsolationDR03() const  { return pfIsolationDR03_; }
+
         const PFIsolation& miniPFIsolation() const { return miniIsolation_; }
+
+        float matchedCaloJetEmEnergy() const { return matchedCaloJetEmEnergy_; }
+        float matchedCaloJetHadEnergy() const { return matchedCaloJetHadEnergy_; }
+
         float dz() const { return dz_; }
+        float dzError() const { return dzError_; }
         float dxy() const { return dxy_; }
+        float dxyError() const { return dxyError_; }
+
+        int fromPV() const { return fromPV_; }
+
+        bool isHighPurityTrack() const 
+          {  return (trackQuality_ & (1 << reco::TrackBase::highPurity)) >> reco::TrackBase::highPurity; }
+        bool isTightTrack() const 
+          {  return (trackQuality_ & (1 << reco::TrackBase::tight)) >> reco::TrackBase::tight; }
+        bool isLooseTrack() const 
+          {  return (trackQuality_ & (1 << reco::TrackBase::loose)) >> reco::TrackBase::loose; }
+
+        const reco::HitPattern& hitPattern() const { return hitPattern_; }
         
+        float dEdxStrip() const { return dEdxStrip_; }
+        float dEdxPixel() const { return dEdxPixel_; }
+
+        const std::vector<EcalChannelStatusCode>& crossedEcalStatus() const { return crossedEcalStatus_; }
+        const std::vector<HcalChannelStatus>& crossedHcalStatus() const { return crossedHcalStatus_; }
+
+
+        float deltaEta() const { return float(deltaEta_)/250*0.5; }
+        float deltaPhi() const { return float(deltaPhi_)/250*0.5; }
```

Comment:
please add some comments (preferably with doxygen formatting ```///``` or ```//!```) to describe what this is.
Other members are somewhat self-explanatory.

Is there a good reason to keep ```/250*0.5``` instead of simply ```/500.f``` ?